### PR TITLE
add binary tree realisation

### PR DESCRIPTION
**Pull request by vladkhard**
close it if you are not interested in this (or maybe you forgot about your work, who knows?). I needed your lib to print my binary tree, but your realisation for list-based trees was inappropriate.

### CODE REVIEW

**Comment by clemtoy** [comment]
Hi,
Thank you for your contribution. I will test your code ASAP. Could you please complete the README ?
Thank you in advance

**Comment by vladkhard** [comment]
@clemtoy i think this is the final version, you could review it when you'll have free time :)

**Comment by clemtoy** [comment]
I don't forget your pull request. I will test as soon as I find a little time. Sorry for the delay.

**Comment by imann24** [comment]
Thanks for adding this feature! Found it super helpful for visual debugging.

File: ppbtree/ppbtree.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+                add(node.right, value)
+
+def print_tree(current_node, nameattr='value', left_child='left', right_child='right', indent='', last='updown'):
+
+    if hasattr(current_node, nameattr):
+        name = lambda node: getattr(node, nameattr)
+    else:
+        name = lambda node: str(node)
+
+    up = getattr(current_node, left_child)
```
Comment (imann24) [inline ppbtree/ppbtree.py:36]: Small nit-pick: personally I favor these reversed `up=right` + `down=left`. Easier for me to visualize if you imagine rotating the tree 90 degrees to the right. But seems a matter of opinion

**Comment by imann24** [comment]
Thanks for adding this one into mainline 💯

**Pull request merged**